onnx/test/reference_evaluator_save_test.py

```diff
@@ -0,0 +1,206 @@
+        onx = self._linear_regression()
+        root = "."
+        path = os.path.join(root, "reference_evaluator_test_save_ref")
+        ref = ReferenceEvaluator(onx, save_intermediate=path)
+        x = np.array([[0, 1], [2, 3]], dtype=np.float32)
+        a = np.array([1, 1], dtype=np.float32)
+        b = np.array([11], dtype=np.float32)
+        ref.run(None, {"X": x, "A": a, "B": b})
+        self.assertTrue(os.path.exists(path))
+        examples = load_model_tests(root, "reference_evaluator_test_save_ref")
+        self.assertEqual(len(examples), 2)
+
+        class NewRef(ReferenceEvaluator):
+            n_inits = 0
+            n_calls = 0
+
+            def __init__(self, *args, **kwargs):
+                NewRef.n_inits += 1
+                ReferenceEvaluator.__init__(  # pylint: disable=non-parent-init-called
+                    self, *args, **kwargs
+                )
+
+            def run(self, *args, **kwargs):
+                NewRef.n_calls += 1
+                return ReferenceEvaluator.run(self, *args, **kwargs)
+
+        new_cls = ReferenceEvaluatorBackend[NewRef]
+        self.assertEqual(new_cls.__name__, "ReferenceEvaluatorBackendNewRef")
+        self.assertTrue(issubclass(new_cls.cls_inference, NewRef))
+
+        backend = create_reference_backend(
+            new_cls, path_to_test=root, kind="reference_evaluator_test_save_ref"
+        )
+        backend.exclude("cuda")
+        tests = backend.tests()
+        names = []
+        for m in dir(tests):
+            if m.startswith("test_"):
+                test = getattr(tests, m)
+                try:
+                    test()
+                    names.append(m)
+                except unittest.case.SkipTest:
+                    continue
+        self.assertEqual(names, ["test_node_0_MatMul_cpu", "test_node_1_Add_cpu"])
+        self.assertEqual(NewRef.n_inits, 2)
+        self.assertEqual(NewRef.n_calls, 2)
+        shutil.rmtree(path)
+
+    @unittest.skipIf(InferenceSession is None, reason="onnxruntime not available")
+    def test_reference_evaluator_onnxruntime_runtime_save(self):
+        onx = self._linear_regression(True)
+        root = "."
+        path = os.path.join(root, "reference_evaluator_test_save_ort")
+        ref = ReferenceEvaluator(onx, save_intermediate=path)
+        x = np.array([[0, 1], [2, 3]], dtype=np.float32)
+        a = np.array([1, 1], dtype=np.float32)
+        b = np.array([11], dtype=np.float32)
+        ref.run(None, {"X": x, "A": a, "B": b})
+        self.assertTrue(os.path.exists(path))
+        examples = load_model_tests(root, "reference_evaluator_test_save_ort")
+        self.assertEqual(len(examples), 2)
+
+        class NewRef(InferenceSession):
+            n_inits = 0
+            n_calls = 0
+
+            def __init__(self, model, *args, providers=None, **kwargs):
+                NewRef.n_inits += 1
+                InferenceSession.__init__(  # pylint: disable=non-parent-init-called
```

Comment:
Why not `super().__init__`?

Reply:
I should. Old habit. It did not exist when I started to code in python and I still don't what it does with multiple inheritance.

Reply:
done